complex/ft_msg.c

```diff
@@ -304,6 +304,183 @@ static int ft_post_send_rma(void)
 	return ret;
 }
 
+int ft_post_send_atomic(void)
+{
+	int ret, i;
+	struct fi_msg_atomic msg;
+	struct fi_rma_ioc rma_iov;
+	size_t iov_count = ft_ctrl.iov_array[ft_tx_ctrl.iov_iter];
+
+	switch (test_info.class_function) {
+		case FT_FUNC_ATOMICV:
+			ft_format_iov(ft_tx_ctrl.iov, iov_count, ft_tx_ctrl.buf,
+				ft_tx_ctrl.count);
+			ft_iov_to_ioc(ft_tx_ctrl.iov, ft_tx_ctrl.ioc, iov_count,
+				ft_tx_ctrl.datatype, ft_tx_ctrl.buf);
```

Comment:
Maybe just add ft_format_ioc, rather than formatting as an iovec, then converting.  (Saying this without having looked at the conversion function yet).